utils/initial_data_preparation.py

```diff
@@ -209,6 +210,39 @@ def is_complete_url(url):
     return df
 
 
+def extract_url(url):
+    if not url:
+        return None, True  # Return None for URL, True for unsuccessful flag
+
+    parsed_url = urlparse(url)
```

Comment:
Very minor change to improve readability; please add a blank line above the comment so the comment is most closely connected with the line of code that follows it (rather than being ambiguous whether it refers to the line of code above, or below, or both).

Reply:
Tried to add a blank line but our pre-commit settings removes it.

Reply:
OK, good to know. I'll investigate the pre-commit configuration.

Reply:
I managed to add the blank line I'd intended, see https://github.com/commercetest/nlnet/pull/76/commits/5f6eb014ce9bf9e79a86c8873724631ee2986f36  
Steps I took:

1. Add the blank line
2. Run the pre-commit checks interactively: `pre-commit run --all-files`
3. `git add utils/initial_data_preparation.py`
4. `git commit ...` to this branch.

Perhaps it'd  worth trying a similar sequence to compare what happened/happens when you do this?

Reply:
I misunderstood where to add the empty line. I was trying to add a blank line for 215 but for the line 219 it works.